lib/fluent/plugin/buffer/file_chunk.rb

```diff
@@ -242,14 +273,24 @@ def file_rename(file, old_path, new_path, callback=nil)
         def create_new_chunk(path, perm)
           @path = self.class.generate_stage_chunk_path(path, @unique_id)
           @meta_path = @path + '.meta'
-          @chunk = File.open(@path, 'wb+', perm)
-          @chunk.set_encoding(Encoding::ASCII_8BIT)
-          @chunk.sync = true
-          @chunk.binmode
-          @meta = File.open(@meta_path, 'wb', perm)
-          @meta.set_encoding(Encoding::ASCII_8BIT)
-          @meta.sync = true
-          @meta.binmode
+          begin
+            @chunk = File.open(@path, 'wb+', perm)
+            @chunk.set_encoding(Encoding::ASCII_8BIT)
+            @chunk.sync = true
+            @chunk.binmode
+          rescue => e
+            raise BufferOverflowError, "can't create buffer file for #{path}. Stop creating buffer files: error = #{e}"
```

Comment:
Does this statement expect "too many open files" or exceptions like that?
Please add a comment about expected exception types - it seems not the direct "buffer overflow" error.